lib/payments/paypal_express/additional_info.rb

```diff
@@ -0,0 +1,135 @@
+require_relative 'api'
+
+# @module FlexCommerce::Payments::PaypalExpress
+module FlexCommerce
+  module Payments
+    module PaypalExpress
+      # @class AdditionalInfo
+      # Address verification service using paypal
+      class AdditionalInfo
+        include ::FlexCommerce::Payments::PaypalExpress::Api
+
+        def initialize(gateway_class: ::ActiveMerchant::Billing::PaypalExpressGateway, shipping_method_model: FlexCommerce::ShippingMethod, options:)
+          self.gateway_class = gateway_class
+          self.token = options[:token]
+          self.shipping_method_model = shipping_method_model
+          self.gateway_details = {}
+        end
+
+        # @method call
+        # 
+        # calculates the meta attributes for shipping method id, 
+        # billing and shipping address
+        # 
+        # @return [FlexCommerce::PaymentAdditionalInfo]
+        def call
+          PaymentAdditionalInfo.new(meta: meta_data, id: SecureRandom.uuid)
```

Comment:
This class needs merging with `PaymentAdditionalInfo` because it's really confusing how we have two things kinda calling one another.